### ISSUE TITLE
[JENKINS-25326] Throttle Concurrent Builds doesn't work when builds are in cloudbees folders

### ISSUE THREAD

**Issue by jenkins-infra-bot** [jenkinsci/throttle-concurrent-builds-plugin#374]
<p>We recently organized our builds into folders, and we noticed that our concurrent build throttling wasn't working. It doesn't work between folders or for builds within the same folder. It only works when it's in the main workspace.</p>

---
<details><summary><i>Originally reported by <a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=gailterman">gailterman</a>, imported from: <a class="original-jira-link" href="https://issues.jenkins.io/browse/JENKINS-25326" target="_blank">Throttle Concurrent Builds doesn't work when builds are in cloudbees folders</a></i></summary>
<i><ul>
<li><b>assignee</b>: <a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=ikedam">ikedam</a>
<li><b>status</b>: Resolved
<li><b>priority</b>: Minor
<li><b>component(s)</b>: throttle-concurrent-builds-plugin
<li><b>label(s)</b>: folders
<li><b>resolution</b>: Fixed
<li><b>resolved</b>: 2016-04-10T09:55:34+00:00
<li><b>votes</b>: 3
<li><b>watchers</b>: 13
<li><b>imported</b>: 20251218-075729
</ul></i>
<details><summary>Raw content of original issue</summary>

<pre>
<p>We recently organized our builds into folders, and we noticed that our concurrent build throttling wasn't working. It doesn't work between folders or for builds within the same folder. It only works when it's in the main workspace.</p></pre>
</details>
</details>
<ul><li><i>environment</i>: <code>jenkins 1.585, throttle concurrent builds plugin 1.8.4, cloudbees folder 4.6.1</code></li></ul>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i><img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/danielbeck.png" title="danielbeck's avatar" /> danielbeck:</i></summary>

<ul>
<li><i>Original <a class="original-jira-link" href="https://issues.jenkins.io/browse/JENKINS-25326?focusedId=214945&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-214945">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p>I confirm the reported behavior on 1.586, 1.8.4, and Folders 4.6.1.</p>

<p>Set up a category 'foo' with 1 maximum concurrent build.
Create two freestyle jobs <a href="https://issues.jenkins.io/browse/JENKINS-25326" title="Throttle Concurrent Builds doesn't work when builds are in cloudbees folders" class="issue-link" data-issue-key="JENKINS-25326"><del>JENKINS-25326</del></a>-one and <a href="https://issues.jenkins.io/browse/JENKINS-25326" title="Throttle Concurrent Builds doesn't work when builds are in cloudbees folders" class="issue-link" data-issue-key="JENKINS-25326"><del>JENKINS-25326</del></a>-two in the same folder 'experiments'. Enable throttling and assign the category 'foo'. Make both of them 'sleep 30' in a shell build step.
Trigger both of them.
They will run in parallel.</p>
</pre>
</details>

</ul>
</details>

<p>I confirm the reported behavior on 1.586, 1.8.4, and Folders 4.6.1.</p>

<p>Set up a category 'foo' with 1 maximum concurrent build.<br/>
Create two freestyle jobs <a href="https://issue-redirect.jenkins.io/issue/25326" title="Throttle Concurrent Builds doesn't work when builds are in cloudbees folders" class="issue-link" data-issue-key="<a class="jira-plain-text-key" href="https://issue-redirect.jenkins.io/issue/25326">JENKINS-25326</a>"><del>JENKINS-25326</del></a>-one and <a href="https://issue-redirect.jenkins.io/issue/25326" title="Throttle Concurrent Builds doesn't work when builds are in cloudbees folders" class="issue-link" data-issue-key="<a class="jira-plain-text-key" href="https://issue-redirect.jenkins.io/issue/25326">JENKINS-25326</a>"><del>JENKINS-25326</del></a>-two in the same folder 'experiments'. Enable throttling and assign the category 'foo'. Make both of them 'sleep 30' in a shell build step.<br/>
Trigger both of them.<br/>
They will run in parallel.</p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i><img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/danielbeck.png" title="danielbeck's avatar" /> danielbeck:</i></summary>

<ul>
<li><i>Original <a class="original-jira-link" href="https://issues.jenkins.io/browse/JENKINS-25326?focusedId=215010&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-215010">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p>There's something weird going on. It seems to only occur with newly created jobs. Jobs that existed when Jenkins was last started are not affected.</p>
</pre>
</details>

</ul>
</details>

<p>There's something weird going on. It seems to only occur with newly created jobs. Jobs that existed when Jenkins was last started are not affected.</p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i><img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/oleg_nenashev.png" title="oleg_nenashev's avatar" /> oleg_nenashev:</i></summary>

<ul>
<li><i>Original <a class="original-jira-link" href="https://issues.jenkins.io/browse/JENKINS-25326?focusedId=215245&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-215245">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p>No response...
AFAIK, it has been fixed in 1.8.2</p>
</pre>
</details>

</ul>
</details>

<p>No response...<br/>
AFAIK, it has been fixed in 1.8.2</p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i><img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/danielbeck.png" title="danielbeck's avatar" /> danielbeck:</i></summary>

<ul>
<li><i>Original <a class="original-jira-link" href="https://issues.jenkins.io/browse/JENKINS-25326?focusedId=215248&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-215248">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p>Oleg: I <b>confirmed</b> the report and added additional information. There's nothing to respond to.</p>
</pre>
</details>

</ul>
</details>

<p>Oleg: I <b>confirmed</b> the report and added additional information. There's nothing to respond to.</p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i><img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/oleg_nenashev.png" title="oleg_nenashev's avatar" /> oleg_nenashev:</i></summary>

<ul>
<li><i>Original <a class="original-jira-link" href="https://issues.jenkins.io/browse/JENKINS-25326?focusedId=215252&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-215252">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p>Yes, my bad. I was going to close <a href="https://issues.jenkins.io/browse/JENKINS-22049" title="propertiesByCategory should not be serialized to hudson.plugins.throttleconcurrents.ThrottleJobProperty.xml" class="issue-link" data-issue-key="JENKINS-22049"><del>JENKINS-22049</del></a>, but selected a wrong tab in the browser.
I'm investigating the issue now</p>
</pre>
</details>

</ul>
</details>

<p>Yes, my bad. I was going to close <a href="https://issue-redirect.jenkins.io/issue/22049" title="propertiesByCategory should not be serialized to hudson.plugins.throttleconcurrents.ThrottleJobProperty.xml" class="issue-link" data-issue-key="<a class="jira-plain-text-key" href="https://issue-redirect.jenkins.io/issue/22049">JENKINS-22049</a>"><del>JENKINS-22049</del></a>, but selected a wrong tab in the browser.<br/>
I'm investigating the issue now</p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i><img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/oleg_nenashev.png" title="oleg_nenashev's avatar" /> oleg_nenashev:</i></summary>

<ul>
<li><i>Original <a class="original-jira-link" href="https://issues.jenkins.io/browse/JENKINS-25326?focusedId=215253&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-215253">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p>Confirmed. It also appears for moved projects.</p>
</pre>
</details>

</ul>
</details>

<p>Confirmed. It also appears for moved projects.</p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i><img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/oleg_nenashev.png" title="oleg_nenashev's avatar" /> oleg_nenashev:</i></summary>

<ul>
<li><i>Original <a class="original-jira-link" href="https://issues.jenkins.io/browse/JENKINS-25326?focusedId=215255&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-215255">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p>This is a regression in the core. The case works well on 1.554.1</p>
</pre>
</details>

</ul>
</details>

<p>This is a regression in the core. The case works well on 1.554.1</p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i><img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/oleg_nenashev.png" title="oleg_nenashev's avatar" /> oleg_nenashev:</i></summary>

<ul>
<li><i>Original <a class="original-jira-link" href="https://issues.jenkins.io/browse/JENKINS-25326?focusedId=215261&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-215261">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p>I cannot reproduce the behavior on 1.585 with installed Matrix Project Plugin.
I tried <a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=danielbeck" class="user-hover" rel="danielbeck">danielbeck</a>'s reproduction steps with additional states (like new category), but nothing helps to reproduce the issue. </p>

<p>Are you sure that Matrix Project plugin was installed?</p>
</pre>
</details>

</ul>
</details>

<p>I cannot reproduce the behavior on 1.585 with installed Matrix Project Plugin.<br/>
I tried <a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=danielbeck" class="user-hover" rel="danielbeck">danielbeck</a>'s reproduction steps with additional states (like new category), but nothing helps to reproduce the issue. </p>

<p>Are you sure that Matrix Project plugin was installed?</p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i><img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/danielbeck.png" title="danielbeck's avatar" /> danielbeck:</i></summary>

<ul>
<li><i>Original <a class="original-jira-link" href="https://issues.jenkins.io/browse/JENKINS-25326?focusedId=215265&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-215265">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p>Oleg: Matrix Project was installed for me.</p>

<p>That said I just set up another pristine instance and can no longer repro. I tried twice from scratch and the plugin works.</p>

<p>Wouldn't resolve this yet though.</p>
</pre>
</details>

</ul>
</details>

<p>Oleg: Matrix Project was installed for me.</p>

<p>That said I just set up another pristine instance and can no longer repro. I tried twice from scratch and the plugin works.</p>

<p>Wouldn't resolve this yet though.</p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i>ulich:</i></summary>

<ul>
<li><i>Original <a class="original-jira-link" href="https://issues.jenkins.io/browse/JENKINS-25326?focusedId=223229&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-223229">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p>I am experiencing the same problem with jenkins 1.596.1 LTS, cloudbees folder plugin 4.7 and Throttle Concurrent Builds Plug-in 1.8.4 (Matrix Project Plugin is installed).</p>

<p>This happens with newly created jobs and jobs that existed before restarting jenkins.</p>

<p>Can I provide any other information?</p>
</pre>
</details>

</ul>
</details>

<p>I am experiencing the same problem with jenkins 1.596.1 LTS, cloudbees folder plugin 4.7 and Throttle Concurrent Builds Plug-in 1.8.4 (Matrix Project Plugin is installed).</p>

<p>This happens with newly created jobs and jobs that existed before restarting jenkins.</p>

<p>Can I provide any other information?</p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i>ikedam:</i></summary>

<ul>
<li><i>Original <a class="original-jira-link" href="https://issues.jenkins.io/browse/JENKINS-25326?focusedId=248734&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-248734">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p>Created a new pull request: <a href="https://github.com/jenkinsci/throttle-concurrent-builds-plugin/pull/37" class="external-link" target="_blank" rel="nofollow noopener">https://github.com/jenkinsci/throttle-concurrent-builds-plugin/pull/37</a></p>
</pre>
</details>

</ul>
</details>

<p>Created a new pull request: <a href="https://github.com/jenkinsci/throttle-concurrent-builds-plugin/pull/37" class="external-link" target="_blank" rel="nofollow noopener">https://github.com/jenkinsci/throttle-concurrent-builds-plugin/pull/37</a></p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i><img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/scm_issue_link.png" title="scm_issue_link's avatar" /> scm_issue_link:</i></summary>

<ul>
<li><i>Original <a class="original-jira-link" href="https://issues.jenkins.io/browse/JENKINS-25326?focusedId=253933&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-253933">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p>Code changed in jenkins
User: ikedam
Path:
 src/main/java/hudson/plugins/throttleconcurrents/ThrottleQueueTaskDispatcher.java
<a href="http://jenkins-ci.org/commit/throttle-concurrent-builds-plugin/09234147a9bf23de0cd40cfa420cb8d4e97a4a6e" class="external-link" target="_blank" rel="nofollow noopener">http://jenkins-ci.org/commit/throttle-concurrent-builds-plugin/09234147a9bf23de0cd40cfa420cb8d4e97a4a6e</a>
Log:
  <span class="error">&#91;FIXED JENKINS-25326&#93;</span> Elevates to SYSTEM when handling throttling in executor threads.</p>


</pre>
</details>

</ul>
</details>

<p>Code changed in jenkins<br/>
User: ikedam<br/>
Path:<br/>
 src/main/java/hudson/plugins/throttleconcurrents/ThrottleQueueTaskDispatcher.java<br/>
<a href="http://jenkins-ci.org/commit/throttle-concurrent-builds-plugin/09234147a9bf23de0cd40cfa420cb8d4e97a4a6e" class="external-link" target="_blank" rel="nofollow noopener">http://jenkins-ci.org/commit/throttle-concurrent-builds-plugin/09234147a9bf23de0cd40cfa420cb8d4e97a4a6e</a><br/>
Log:<br/>
  <span class="error">&#91;FIXED <a class="jira-plain-text-key" href="https://issue-redirect.jenkins.io/issue/25326">JENKINS-25326</a>&#93;</span> Elevates to SYSTEM when handling throttling in executor threads.</p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i><img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/scm_issue_link.png" title="scm_issue_link's avatar" /> scm_issue_link:</i></summary>

<ul>
<li><i>Original <a class="original-jira-link" href="https://issues.jenkins.io/browse/JENKINS-25326?focusedId=253934&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-253934">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p>Code changed in jenkins
User: ikedam
Path:
 pom.xml
 src/test/java/hudson/plugins/throttleconcurrents/ThrottleIntegrationTest.java
<a href="http://jenkins-ci.org/commit/throttle-concurrent-builds-plugin/5b5728a069df14df1ce48377682b74ed338605df" class="external-link" target="_blank" rel="nofollow noopener">http://jenkins-ci.org/commit/throttle-concurrent-builds-plugin/5b5728a069df14df1ce48377682b74ed338605df</a>
Log:
  <a href="https://issues.jenkins.io/browse/JENKINS-25326" title="Throttle Concurrent Builds doesn't work when builds are in cloudbees folders" class="issue-link" data-issue-key="JENKINS-25326"><del>JENKINS-25326</del></a> Reverted changes required for the test to reproduce <a href="https://issues.jenkins.io/browse/JENKINS-25326" title="Throttle Concurrent Builds doesn't work when builds are in cloudbees folders" class="issue-link" data-issue-key="JENKINS-25326"><del>JENKINS-25326</del></a>.</p>


</pre>
</details>

</ul>
</details>

<p>Code changed in jenkins<br/>
User: ikedam<br/>
Path:<br/>
 pom.xml<br/>
 src/test/java/hudson/plugins/throttleconcurrents/ThrottleIntegrationTest.java<br/>
<a href="http://jenkins-ci.org/commit/throttle-concurrent-builds-plugin/5b5728a069df14df1ce48377682b74ed338605df" class="external-link" target="_blank" rel="nofollow noopener">http://jenkins-ci.org/commit/throttle-concurrent-builds-plugin/5b5728a069df14df1ce48377682b74ed338605df</a><br/>
Log:<br/>
  <a href="https://issue-redirect.jenkins.io/issue/25326" title="Throttle Concurrent Builds doesn't work when builds are in cloudbees folders" class="issue-link" data-issue-key="<a class="jira-plain-text-key" href="https://issue-redirect.jenkins.io/issue/25326">JENKINS-25326</a>"><del>JENKINS-25326</del></a> Reverted changes required for the test to reproduce <a href="https://issue-redirect.jenkins.io/issue/25326" title="Throttle Concurrent Builds doesn't work when builds are in cloudbees folders" class="issue-link" data-issue-key="<a class="jira-plain-text-key" href="https://issue-redirect.jenkins.io/issue/25326">JENKINS-25326</a>"><del>JENKINS-25326</del></a>.</p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i><img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/scm_issue_link.png" title="scm_issue_link's avatar" /> scm_issue_link:</i></summary>

<ul>
<li><i>Original <a class="original-jira-link" href="https://issues.jenkins.io/browse/JENKINS-25326?focusedId=253936&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-253936">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p>Code changed in jenkins
User: Oleg Nenashev
Path:
 pom.xml
 src/main/java/hudson/plugins/throttleconcurrents/ThrottleQueueTaskDispatcher.java
 src/test/java/hudson/plugins/throttleconcurrents/ThrottleIntegrationTest.java
 src/test/java/hudson/plugins/throttleconcurrents/ThrottleQueueTaskDispatcherTest.java
<a href="http://jenkins-ci.org/commit/throttle-concurrent-builds-plugin/c823d8d9284aa6d0e645b7c2a7702a4d1d354e2c" class="external-link" target="_blank" rel="nofollow noopener">http://jenkins-ci.org/commit/throttle-concurrent-builds-plugin/c823d8d9284aa6d0e645b7c2a7702a4d1d354e2c</a>
Log:
  Merge pull request #41 from oleg-nenashev/feature/<a href="https://issues.jenkins.io/browse/JENKINS-25326" title="Throttle Concurrent Builds doesn't work when builds are in cloudbees folders" class="issue-link" data-issue-key="JENKINS-25326"><del>JENKINS-25326</del></a>_ExecutorPermission</p>

<p><a href="https://issues.jenkins.io/browse/JENKINS-25326" title="Throttle Concurrent Builds doesn't work when builds are in cloudbees folders" class="issue-link" data-issue-key="JENKINS-25326"><del>JENKINS-25326</del></a> - Elevate user to system during build throttling</p>


<p>Compare: <a href="https://github.com/jenkinsci/throttle-concurrent-builds-plugin/compare/e1aeb176b649...c823d8d9284a" class="external-link" target="_blank" rel="nofollow noopener">https://github.com/jenkinsci/throttle-concurrent-builds-plugin/compare/e1aeb176b649...c823d8d9284a</a></p>

</pre>
</details>

</ul>
</details>

<p>Code changed in jenkins<br/>
User: Oleg Nenashev<br/>
Path:<br/>
 pom.xml<br/>
 src/main/java/hudson/plugins/throttleconcurrents/ThrottleQueueTaskDispatcher.java<br/>
 src/test/java/hudson/plugins/throttleconcurrents/ThrottleIntegrationTest.java<br/>
 src/test/java/hudson/plugins/throttleconcurrents/ThrottleQueueTaskDispatcherTest.java<br/>
<a href="http://jenkins-ci.org/commit/throttle-concurrent-builds-plugin/c823d8d9284aa6d0e645b7c2a7702a4d1d354e2c" class="external-link" target="_blank" rel="nofollow noopener">http://jenkins-ci.org/commit/throttle-concurrent-builds-plugin/c823d8d9284aa6d0e645b7c2a7702a4d1d354e2c</a><br/>
Log:<br/>
  Merge pull request #41 from oleg-nenashev/feature/<a href="https://issue-redirect.jenkins.io/issue/25326" title="Throttle Concurrent Builds doesn't work when builds are in cloudbees folders" class="issue-link" data-issue-key="<a class="jira-plain-text-key" href="https://issue-redirect.jenkins.io/issue/25326">JENKINS-25326</a>"><del>JENKINS-25326</del></a>_ExecutorPermission</p>

<p><a href="https://issue-redirect.jenkins.io/issue/25326" title="Throttle Concurrent Builds doesn't work when builds are in cloudbees folders" class="issue-link" data-issue-key="<a class="jira-plain-text-key" href="https://issue-redirect.jenkins.io/issue/25326">JENKINS-25326</a>"><del>JENKINS-25326</del></a> - Elevate user to system during build throttling</p>


<p>Compare: <a href="https://github.com/jenkinsci/throttle-concurrent-builds-plugin/compare/e1aeb176b649...c823d8d9284a" class="external-link" target="_blank" rel="nofollow noopener">https://github.com/jenkinsci/throttle-concurrent-builds-plugin/compare/e1aeb176b649...c823d8d9284a</a></p>
